packages/pds/src/sequencer/sequencer-leader.ts

```diff
@@ -0,0 +1,127 @@
+import { DisconnectError } from '@atproto/xrpc-server'
+import { jitter, wait } from '@atproto/common'
+import { Leader } from '../db/leader'
+import { seqLogger as log } from '../logger'
+import Database from '../db'
+
+export const SEQUENCER_LEADER_ID = 1100
+
+export class SequencerLeader {
+  leader: Leader
+
+  destroyed = false
+  polling = false
+  queued = false
+  lastSeq: number
+
+  constructor(public db: Database) {
+    const lockId = this.db.getSchemaAdvisoryLockId(SEQUENCER_LEADER_ID)
+    this.leader = new Leader(lockId, this.db)
+  }
+
+  nextSeqVal(): number {
+    this.lastSeq++
+    return this.lastSeq
+  }
+
+  async run() {
+    if (this.db.dialect === 'sqlite') return
+
+    const res = await this.db.db
+      .selectFrom('repo_seq')
+      .select('seq')
+      .orderBy('seq', 'desc')
+      .limit(1)
+      .executeTakeFirst()
+    this.lastSeq = res?.seq ?? 0
+
+    while (!this.destroyed) {
+      try {
+        const { ran } = await this.leader.run(async ({ signal }) => {
+          const seqListener = () => {
+            if (this.polling) {
+              this.queued = true
+            } else {
+              this.polling = true
+              this.pollDb()
+            }
+          }
+          this.db.channels.new_repo_event.addListener('message', seqListener)
+          await new Promise<void>((resolve) => {
+            signal.addEventListener('abort', () => {
```

Comment:
If you want to be really defensive you could check if it's already aborted here with `signal.aborted` too.  Not a super practical issue, but could turn-up in tests.